### PR TITLE
Add type annotations to part of the public API

### PR DESCRIPTION
**Pull request by basnijholt**

### CODE REVIEW

File: qiskit_ionq/__init__.py [hunk 1]
```diff
@@ -26,8 +26,6 @@
 
 """Provider for IonQ backends"""
 
-import warnings
```
Comment (basnijholt) [inline qiskit_ionq/__init__.py:29]: Was unused.

**Review by Cynocracy — APPROVED** [review]

**Comment by Cynocracy** [comment]
Looks great, thanks for adding @basnijholt !

File: setup.py [hunk 2]
```diff
@@ -78,7 +78,7 @@
         "Topic :: Scientific/Engineering",
     ],
     keywords="qiskit sdk quantum",
-    python_requires=">=3.7",
+    python_requires=">=3.9",
```
Comment (basnijholt) [inline setup.py:81]: Python 3.7 is already EOL and 3.8 will be at the end of this month.

Comment (Cynocracy) [reply setup.py:81]: makes sense, ty again!

**Pull request merged**